x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/InlineStats.java

```diff
@@ -119,31 +87,51 @@ public String commandName() {
 
     @Override
     public boolean expressionsResolved() {
-        return Resolvables.resolved(groupings) && Resolvables.resolved(aggregates);
+        return aggregate.expressionsResolved();
     }
 
     @Override
     public List<Attribute> output() {
         if (this.lazyOutput == null) {
-            List<NamedExpression> addedFields = new ArrayList<>();
-            AttributeSet set = child().outputSet();
+            this.lazyOutput = mergeOutputAttributes(aggregate.output(), aggregate.child().output());
+        }
+        return lazyOutput;
+    }
+
+    // TODO: in case of inlinestats, the join key is always the grouping
+    private JoinConfig joinConfig() {
+        List<Expression> groupings = aggregate.groupings();
+        List<Attribute> namedGroupings = new ArrayList<>(groupings.size());
+        for (Expression g : groupings) {
+            namedGroupings.add(Expressions.attribute(g));
+        }
 
```

Comment:
This part needs to be revisited and JoinConfig refactored to better incorporate the field resolution.